### PR TITLE
Fix CD tool argument building for arguments containing spaces

### LINKED ISSUE #72: Can't generate an SBOM for directories containing a space
I execute this command on Windows (doesn't matter if I use cmd or PowerShell):

```
C:\Temp\sbom-tool.exe generate -b "C:\src\core\Build\Service with spaces" -bc "C:\src\core\Solutions\service with spaces\src\" -pn "Foo bar services" -pv "1.0.0.0" -nsb "https://example.com"
```
The tool runs and creates a manifest, however no depedencies are found.

Program output:
```
[INFO] Log file: C:\Users\username\AppData\Local\Temp\GovCompDisc_Log_20220722135038.log
[INFO] Run correlation id: 77b85f03-caa7-47b4-bc71-94e9e4b003c1

[INFO] Attempting to load default detectors
[INFO] 19 detectors were found in Microsoft.ComponentDetection.Detectors

[WARN] Provided search path C:\Users\username\AppData\Local\Temp\.net\sbom-tool\lpvnvva1.mst\Plugins does not exist.


[INFO] Finding components...
[INFO] No instructions received to scan docker images.
[ERROR] Root directory doesn't exist: C:\src\core\Solutions\service
```

If I run the tool on a directory without spaces, everything works fine.

### PR DESCRIPTION
**Pull request by ByAgenT**
Resolves #72

### CODE REVIEW

File: src/Microsoft.Sbom.Api/Utils/ComponentDetectionCliArgumentBuilder.cs [hunk 1]
```diff
@@ -85,7 +85,7 @@ public ComponentDetectionCliArgumentBuilder Scan()
 
         public ComponentDetectionCliArgumentBuilder AddDetectorArg(string name, string value)
         {
-            detectorArgs.TryAdd(name, value);
+            detectorArgs[name] = value;
```
Comment (edgarrs) [inline src/Microsoft.Sbom.Api/Utils/ComponentDetectionCliArgumentBuilder.cs:88]: >detectorArgs [](http://example.com/codeflow?start=12&length=12)

Do want to revert this recent change?

File: src/Microsoft.Sbom.Api/Utils/ComponentDetectionCliArgumentBuilder.cs [hunk 2]
```diff
@@ -203,5 +203,15 @@ public ComponentDetectionCliArgumentBuilder ParseAndAddArgs(string args)
 
             return this;
         }
+
+        private string AsArgumentValue(string arg)
+        {
+            if (arg.Contains(' '))
+            {
+                return $"\"{arg}\"";
+            }
```
Comment (edgarrs) [inline src/Microsoft.Sbom.Api/Utils/ComponentDetectionCliArgumentBuilder.cs:212]: This work regardless of the OS, right? Did you test you changes in Mac\Linux\Windows?

File: test/Microsoft.Sbom.Api.Tests/Utils/ComponentDetectionCliArgumentBuilderTests.cs [hunk 3]
```diff
@@ -154,5 +172,47 @@ public void Build_WithInvalidArg()
 
             builder.Build();
         }
+
+        [TestMethod]
+        public void Build_WithSpacesSourceDirectory()
```
Comment (edgarrs) [inline test/Microsoft.Sbom.Api.Tests/Utils/ComponentDetectionCliArgumentBuilderTests.cs:177]: >Build_WithSpacesSourceDirectory [](http://example.com/codeflow?start=20&length=31)

Love this

**Review by edgarrs — APPROVED** [review]
Please take a look at my comments

File: src/Microsoft.Sbom.Api/Utils/ComponentDetectionCliArgumentBuilder.cs (hunk 1 above)
Comment (ByAgenT) [reply src/Microsoft.Sbom.Api/Utils/ComponentDetectionCliArgumentBuilder.cs:88]: No, it's not necessary. Last PR fixed actual issue, but this change is just an "improvement" to make things more logical and matching my expectation as a consumer of this class :)

File: src/Microsoft.Sbom.Api/Utils/ComponentDetectionCliArgumentBuilder.cs (hunk 2 above)
Comment (ByAgenT) [reply src/Microsoft.Sbom.Api/Utils/ComponentDetectionCliArgumentBuilder.cs:212]: Yea this is platform independent. But I'll do a sanity check on Ubuntu before merging

File: src/Microsoft.Sbom.Api/Utils/ComponentDetectionCliArgumentBuilder.cs [hunk 4]
```diff
@@ -203,5 +203,15 @@ public ComponentDetectionCliArgumentBuilder ParseAndAddArgs(string args)
 
             return this;
         }
+
+        private string AsArgumentValue(string arg)
```
Comment (rgl) [inline src/Microsoft.Sbom.Api/Utils/ComponentDetectionCliArgumentBuilder.cs:207]: this implementation is scary simple. dotnet does not yet have a proper api to call external processes?

**Pull request merged**